src/aiodynamo/credentials.py

```diff
@@ -307,7 +307,9 @@ async def fetch_metadata(self, http: HTTP) -> Metadata:
                 timeout=self.timeout,
             )
         ).decode("utf-8")
-        credentials_url = role_url.join(URL(role))
+        credentials_url = self.base_url.with_path(
+            f"/latest/meta-data/iam/security-credentials/{role}"
```

Comment:
maybe unrelated:
I went to check what `role` might be... just in case it's smth like `../../../path-to-exfiltrate-data`
And I realised that `(aiohttp|httpx).get()` protects against status codes >=400, but not 302 or 201 or 204.
So, hypothetically, `role` could be empty or even random text.
I dunno what could break in that case...

Reply:
🤔 if you compromised the instance metadata endpoint somehow, you can already hijack the first request (`"/latest/meta-data/iam/security-credentials/"`) so I don't think this changes much.